### ISSUE TITLE
SVG parsing/drawing

### ISSUE THREAD

**Issue by suomilapsi** [kangax/fabric.js#2696]
Hello,

SVG inner object is completely black at fabricjs canvas where as browsers display it with details.
https://jsfiddle.net/suomilapsi/hmLcah9u/2/

Any idea how to fix this?

Best regards, suomilapsi

**Comment by suomilapsi** [comment]
SVG stroke seems what is not handled correctly e.g. fill-opacity="0" stroke="#000000" stroke-width="8".

**Comment by asturur** [comment]
https://jsfiddle.net/hmLcah9u/3/

 does not work also with  loadFromUrl ( some user spoke about some differencies)

i will look at it, it looks like a bug.

**Comment by asturur** [comment]
error looks like here, if we do not have fill definied but fill-opacity we skip the properties instead of defining the default black.

```
  /**
   * @private
   * @param {Object} attributes Array of attributes to parse
   */
  function _setStrokeFillOpacity(attributes) {
    for (var attr in colorAttributes) {

      if (!attributes[attr] || typeof attributes[colorAttributes[attr]] === 'undefined') {
        continue;
      }

      if (attributes[attr].indexOf('url(') === 0) {
        continue;
      }

      var color = new fabric.Color(attributes[attr]);
      attributes[attr] = color.setAlpha(toFixed(color.getAlpha() * attributes[colorAttributes[attr]], 2)).toRgba();
    }
    return attributes;
  }
```

**Comment by asturur** [comment]
```javascript
  /**
   * @private
   * @param {Object} attributes Array of attributes to parse
   */
  function _setStrokeFillOpacity(attributes) {
    for (var attr in colorAttributes) {

      if (typeof attributes[colorAttributes[attr]] === 'undefined') {
        continue;
      }

      if (!attributes[attr]) {
        if (!fabric.Object.prototype[attr]) {
          continue;
        }
        attributes[attr] = fabric.Object.prototype[attr];
      }

      if (attributes[attr].indexOf('url(') === 0) {
        continue;
      }

      var color = new fabric.Color(attributes[attr]);
      attributes[attr] = color.setAlpha(toFixed(color.getAlpha() * attributes[colorAttributes[attr]], 2)).toRgba();
    }
    return attributes;
  }
```

this is the cleanest solution i can think of.
What do you think @kangax?

**Comment by kangax** [comment]
@asturur seems fine

**Comment by suomilapsi** [comment]
I can confirm that all svgs that I have with this issue work with above code. Thanks for fast analysis and fix.

**Comment by asturur** [comment]
we closed this with #2703
